contracts/near/eth-types/src/lib.rs

```diff
@@ -299,6 +298,11 @@ impl From<BlockHeaderPreLondon> for BlockHeader {
 
 impl BorshDeserialize for BlockHeader {
     fn deserialize(buf: &mut &[u8]) -> std::io::Result<Self> {
```

Comment:
@sept-en  Just note about this. We noticed that if deserialization for London block fails, one for PreLondon fails as well. I guess something gets mutated in the first call. We got around that using feature flag, but just something to be aware of.